packages/server/lib/util/blacklist.coffee

```diff
@@ -0,0 +1,37 @@
+_ = require("lodash")
+url = require("url")
+minimatch = require("minimatch")
+
+DEFAULT_PORTS = ["443", "80"]
+
+stripProtocolAndDefaultPorts = (urlToCheck) ->
+  ## grab host which is 'hostname:port' only
+  { host, hostname, port } = url.parse(urlToCheck)
+
+  ## if we have a default port for 80 or 443
+  ## then just return the hostname
+  if port in DEFAULT_PORTS
+    return hostname
+
+  ## else return the host
+  return host
+
+matches = (urlToCheck, blacklistHosts) ->
```

Comment:
I would check `blackListHosts` type here - if type is not string or array of strings, something went wrong. Throw

Reply:
also, what does this return? needs a jsdoc block I feel, because this is so reusable